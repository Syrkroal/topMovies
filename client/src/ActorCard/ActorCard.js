import "./ActorCard.css";

const imgPath = "https://image.tmdb.org/t/p/w500/";

function ActorCard(props) {
    const actor = props.actor;

    return (
        <div className="actor-container">
            <img
                className="actor-img"
                src={imgPath + actor.profile_path}
                alt="Poster"
            />
            <div className="actor-name-container">
                <div className="actor-character">{actor.character}</div>
                <div className="actor-name">{actor.name}</div>
            </div>
        </div>
    );
}

export default ActorCard;
