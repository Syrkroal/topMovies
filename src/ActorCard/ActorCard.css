.actor-img {
    height: 220px;
    width: 146px;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
}

.actor-container {
    border-radius: 10px;
    background-color: #141414;
    cursor: pointer;
    margin: 10px;
    display: flex;
    flex-direction: column;
    color: white;
    font-size: 20px;
}

.actor-character {
    font-weight: bold;
    font-size: 16px;
}

.actor-name {
    font-size: 16px;
}

.actor-name-container {
    margin: 10px;
}
