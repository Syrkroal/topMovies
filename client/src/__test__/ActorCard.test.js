import React from "react";
import Enzyme, { shallow } from "enzyme";
import ActorCard from "../ActorCard/ActorCard";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("ActorCard to match snapshot", () => {
    const actor = {
        profile_path: "path",
        name: "name",
        character: "character",
    };

    const container = shallow(<ActorCard actor={actor} />);
    it("should match the snapshot", () => {
        expect(container.html()).toMatchSnapshot();
    });
});
