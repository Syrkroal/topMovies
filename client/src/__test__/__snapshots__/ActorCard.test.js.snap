// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorCard to match snapshot should match the snapshot 1`] = `"<div class=\\"actor-container\\"><img class=\\"actor-img\\" src=\\"https://image.tmdb.org/t/p/w500/path\\" alt=\\"Poster\\"/><div class=\\"actor-name-container\\"><div class=\\"actor-character\\">character</div><div class=\\"actor-name\\">name</div></div></div>"`;
